spec/unit/ace/transport_app_spec.rb

```diff
@@ -308,6 +318,79 @@ def app
       end
     end
 
+    context 'when given flag debug' do
+      it 'returns 200 with success' do
+        execute_catalog_body[:compiler][:debug] = true
+
+        expect(psettings).to receive(:[]=).with(:noop, false)
+        expect(psettings).to receive(:[]=).with(:log_level, :debug)
+        expect(psettings).to receive(:[]=).with(:trace, false)
+        expect(psettings).to receive(:[]=).with(:evaltrace, false)
+
+        # handle the reset of log_level
+        expect(psettings).to receive(:[]=).with(:log_level, :wibble)
+
+        post '/execute_catalog', JSON.generate(execute_catalog_body), 'CONTENT_TYPE' => 'text/json'
+
+        expect(configurer).to have_received(:run)
+        expect(last_response.errors).to match(/\A\Z/)
+        expect(last_response).to be_ok
+        expect(last_response.status).to eq(200)
+      end
+    end
+
+    context 'when given flag noop' do
+      it 'returns 200 with success' do
+        execute_catalog_body[:compiler][:noop] = true
+
+        expect(psettings).to receive(:[]=).with(:noop, true)
+        expect(psettings).to receive(:[]=).with(:trace, false)
+        expect(psettings).to receive(:[]=).with(:evaltrace, false)
```

Comment:
for completeness sake please add `expect(psettings).not_to receive(:[]=).with(:log_level, :debug)` here and below.

Reply:
Done